src/authorized_keys_d/authorized_keys_d.go

```diff
@@ -304,9 +305,12 @@ func (akd *SSHAuthorizedKeysDir) Disable(name string) error {
 // Add adds the supplied key at name.
 // replace enables replacing keys already existing at name.
 // force enables adding keys to a disabled name, enabling it in the process.
+// Names "..", ".", and anything containing "/" are disallowed.
 func (akd *SSHAuthorizedKeysDir) Add(name string, keys []byte, replace, force bool) error {
-	// TODO FIXME(vc): name can contain anything
-	// either add escaping, or place restrictions on its contents.
+	if name == ".." || name == "." || strings.Contains(name, "/") {
```

Comment:
I would ban all names starting with `.` and similarly ignore all files in the directory starting with `.` in order to avoid things like vim `.swp` files and other potential noise.